tensorflow/compiler/xla/status_macros.h

```diff
@@ -196,6 +196,22 @@ class StatusAdaptorForMacros {
 #define TF_STATUS_MACROS_CONCAT_NAME(x, y) TF_STATUS_MACROS_CONCAT_IMPL(x, y)
 #define TF_STATUS_MACROS_CONCAT_IMPL(x, y) x##y
 
+#ifdef _MSC_VER
+
+#define TF_ASSIGN_OR_RETURN(lhs, rexpr)                                   \
+  TF_ASSIGN_OR_RETURN_IMPL(TF_STATUS_MACROS_CONCAT_NAME(_status_or_value, \
+                                                        __COUNTER__),     \
+                           lhs, rexpr)
+
+#define TF_ASSIGN_OR_RETURN_IMPL(statusor, lhs, rexpr) \
```

Comment:
I think this is identical to TF_ASSIGN_OR_RETURN_IMPL for the non-msc case.  Let's refactor the code to minimize the differences.